integrations/octopus/main.py

```diff
@@ -48,25 +40,57 @@ async def setup_application() -> None:
         )
         return
     octopus_client = await init_client()
-    existing_subscriptions = await octopus_client.get_webhook_subscriptions()
-    existing_webhook_uris = {
-        subscription.get("EventNotificationSubscription", {}).get("WebhookURI")
-        for subscription in existing_subscriptions
-    }
-    webhook_uri = f"{app_host}/integration/webhook"
-    if webhook_uri in existing_webhook_uris:
-        logger.info(f"Webhook already exists with URI: {webhook_uri}")
-    else:
-        await octopus_client.create_webhook_subscription(app_host)
-        logger.info(f"Webhook created with URI: {webhook_uri}")
+    async for spaces in octopus_client.get_all_spaces():
+        for space in spaces:
+            async for subscriptions in octopus_client.get_webhook_subscriptions(
+                space.get("Id")
+            ):
+                existing_webhook_uris = {
+                    subscription.get("EventNotificationSubscription", {}).get(
+                        "WebhookURI"
+                    )
+                    for subscription in subscriptions
+                }
+                existing_webhook_names = [
+                    subscription.get("Name", "") for subscription in subscriptions
+                ]
```

Comment:
is there a reason we are checking for webhook name instead of using the URL?

Reply:
We need to track both, unlike other webhooks I have worked with, name is required and must be unique when creating octopus subscription and the url is not required since the subscription can be an email subscription too. Tracking url let us know if that particular url has been previously created and tracking the name if the url does not exist only helps avoid making an api call that will definitely throw a 400

Reply:
This situation will only happen in rare cases whereby a customer decides to switch hosts, like maybe point a different subdomain at the integration